opm/autodiff/FullyImplicitBlackoilSolver_impl.hpp

```diff
@@ -286,7 +296,7 @@ namespace detail {
 
             detectNewtonOscillations(residual_norms_history, it, relaxRelTol(), isOscillate, isStagnate);
 
-            if (isOscillate) {
+            if (isOscillate && verbosity_) {
```

Comment:
It looks like the `verbosity_` is used as a flag for "are we on rank 0" here. I am not sure if changing omega should be done on all nodes or just rank 0, but even so, that should be written explicitly instead of using `verbosity_`. That should be reserved for output only. If you want to use a single flag, it should be called `on_rank_0` instead (otherwise I prefer `terminal_output`).